src/eventbridge/rule.ts

```diff
@@ -0,0 +1,164 @@
+import { aws_events } from "aws-cdk-lib";
+import { Construct } from "constructs";
+import { EventBus } from "./bus";
+import { FunctionDecl } from "../declaration";
+import { synthesizeEventPattern } from "./eventpattern";
+import { Function } from "../function";
+import { EventBusRuleInput } from "./types";
+import { EventTransformFunction, EventBusTransform } from "./transform";
+import {
+  EventBusTargetProps,
+  EventBusTargetResource,
+  LambdaTargetProps,
+  pipe,
+} from "./target";
+
+/**
+ * A function interface used by the {@link EventBus}'s when function to generate a rule.
+ *
+ * event is every event sent to the bus to be filtered. This argument is optional.
+ */
+export type EventPredicateFunction<
+  E extends EventBusRuleInput = EventBusRuleInput<any>
+> = (event: E) => boolean;
+
+/**
+ * Represents a set of events filtered by the when predicate using event bus's EventPatterns.
+ * https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-patterns.html
+ *
+ * See {@link EventBus}.when for more details on filtering events.
```

Comment:
This should be `{@link EventBus.when}` - notice how if you change it to this, you can use jump-to-definition to go straight to that method.

Reply:
Should we provide some extra documentation on this class? 

> To create EventBusRules, use the {@link EventBus.when} helper instead of instantiating this class directly.
> ```
> (single example with .when, .map, .transform)
> ```